opencsp/common/lib/cv/AbstractFiducials.py

```diff
@@ -0,0 +1,152 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+import matplotlib.axes
+import matplotlib.pyplot as plt
+import numpy as np
+
+import opencsp.common.lib.geometry.Pxy as p2
+import opencsp.common.lib.geometry.RegionXY as reg
+import opencsp.common.lib.geometry.Vxyz as v3
+import opencsp.common.lib.render.figure_management as fm
+import opencsp.common.lib.render_control.RenderControlPointSeq as rcps
+import opencsp.common.lib.tool.log_tools as lt
+
+
+class AbstractFiducials(ABC):
+    def __init__(self, style=None, pixels_to_meters: Callable[[p2.Pxy], v3.Vxyz] = None):
+        """
+        A collection of markers (such as an ArUco board) that is used to orient the camera relative to observed objects
+        in the scene. It is suggested that each implementing class be paired with a complementary locator method or
+        SpotAnalysisImageProcessor.
+
+        Parameters
+        ----------
+        style : RenderControlPointSeq, optional
+            How to render this fiducial when using the defaul render_to_plot() method. By default rcps.default().
+        pixels_to_meters : Callable[[p2.Pxy], v3.Vxyz], optional
+            Conversion function to get the physical point in space for the given x/y position information. Used in the
+            default self.scale implementation. Defaults to 1 meter per pixel.
+        """
+        self.style = style if style is not None else rcps.default()
+        self.pixels_to_meters = pixels_to_meters
+
+    @abstractmethod
+    def get_bounding_box(self, index=0) -> reg.RegionXY:
+        """The X/Y bounding box(es) of this instance, in pixels."""
+
+    @property
+    @abstractmethod
+    def origin(self) -> p2.Pxy:
+        """The origin point(s) of this instance, in pixels."""
+
+    @property
+    @abstractmethod
+    def orientation(self) -> v3.Vxyz:
+        """The orientation(s) of this instance, in radians. This is relative to
+        the source image, where x is positive to the right, y is positive down,
+        and z is positive in (away from the camera)."""
```

Comment:
That is a brilliant idea! Thanks! 🧠

If you don't mind, please take a look at the updated description and let me know what you think https://github.com/sandialabs/OpenCSP/commit/2568e2d5ee86c5a94786f6e4abb18afafe3399c0:

```python
"""
The orientation of the normal vector(s) of this instance.
This is relative to the orthorectified source image, where x is positive
to the right, y is positive down, and z is positive in (away from the
camera).

This can be used to describe the forward transformation from the
camera's perspective. For example, an aruco marker whose origin is in
the center of the image and is facing towards the camera could have the
orientation::

    Rotation.from_euler('y', np.pi)

If that same aruco marker was also placed upside down, then it's
orientation could be::

    Rotation.from_euler(
        'yz',
        [ [np.pi, 0],
            [0,     np.pi] ]
    )
"""
```

Reply:
I think that description is overall good! I'm not sure I understand what is meant by:
```
relative to the orthorectified source image
```
Is this the image on the camera sensor? If so, why is this orthorectified? If it is the camera or the target coordinates, I would just expect "camera" or "target."